CONTRIBUTING.md

```diff
@@ -72,6 +72,18 @@ _Before_ submitting a pull request, please make sure the following is done…
 
 1.  If you haven't already, complete the CLA.
 
+#### Testing
+
+Code that is written needs to be tested to ensure that it achieves the desired behaviour. Tests either fall into a unit test or an integration test.
+
+##### Unit tests
+
+Some of the packages within jest have a `__tests__` directory. This is where unit tests reside in. If the scope of your work only requires a unit test, this is where you will write it in. Tests here usually don't require much if any setup.
+
+##### Integration tests
+
+There will be situations however where the work you have done cannot be tested alone using unit tests. In situations like this, you should write an integration test for your code. The integration tests reside within the `e2e` directory. Within this directory, there is a `__tests__` directory. This is where you will write the integration test itself. The tests within this directory execute jest itself using `runJest.js` and assertions are usually made on one if not all the output of the following `status`, `stdout` and `stderr`. The other sub directories within the `e2e` directory are where you will write the test files that jest will run for your integration tests. Feel free to take a look at any of the tests in the `__tests__` directory within `e2e` to have a better sense of how it is currently being done.
```

Comment:
Maybe add that you can run the integration test to inspect that it behaves like you want by running `../../jest` while inside it. Useful when debugging a failing one as well.

E.g.

```sh-session
$ cd e2e/clear-cache
$ ../../jest
 PASS  __tests__/clear_cache.test.js
  ✓ stub (4ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        1.225s
Ran all test suites.
```

Reply:
go for it!